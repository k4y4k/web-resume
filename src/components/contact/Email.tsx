import * as React from 'react'
import { FiMail } from '@react-icons/all-files/fi/FiMail'

interface EmailTypes {
  email: string
}

const Email = ({ email }: EmailTypes): JSX.Element | null => {
  if (email === '') return null

  return (
    <a data-testid='contactEmail' href={`mailto:${email}`}>
      <FiMail /> {email}
    </a>
  )
}

export default Email
