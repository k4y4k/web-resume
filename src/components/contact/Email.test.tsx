import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Email from './Email'

describe('<Email/>', () => {
  test('Handles no data case properly', () => {
    render(<Email email='' />)
    const email = screen.getByText('Error: No Email')

    expect(email).not.toBeFalsy()
  })

  test('displays email, properly formatted', () => {
    render(<Email email='kayak@example.com' />)

    const regex = /\S+@\S+\.\S+/
    const email = screen.getByText(regex)

    expect(email).not.toBeFalsy()
  })

  test('mailto: link', () => {
    render(<Email email='kayak@example.com' />)

    const regex = /\S+@\S+\.\S+/
    const email = screen.getByText(regex)

    expect(email).toHaveAttribute('href', 'mailto:kayak@example.com')
  })
})
