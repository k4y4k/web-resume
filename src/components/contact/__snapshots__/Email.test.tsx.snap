// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Email/> Handles no data case properly 1`] = `null`;

exports[`<Email/> displays email, properly formatted 1`] = `
<a
  data-testid="contactEmail"
  href="mailto:kayak@example.com"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
    />
    <polyline
      points="22,6 12,13 2,6"
    />
  </svg>
   
  kayak@example.com
</a>
`;

exports[`<Email/> mailto: link 1`] = `
<a
  data-testid="contactEmail"
  href="mailto:kayak@example.com"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
    />
    <polyline
      points="22,6 12,13 2,6"
    />
  </svg>
   
  kayak@example.com
</a>
`;
